### PR TITLE
V3 update rma status

### PR DESCRIPTION
**Pull request by filafb**
_Disclaimer: It got bigger than I imagined, sorry 😬._

This PR connects front end with [this resolver](https://github.com/vtex-apps/return-app/pull/105).

It brings the component to update the request status, including the page to verify the items while request is on status `pendingVerification`.

The most important components to review are:

1. __UpdateRequestStatus.tsx__
- Manages the input of the new status to be updated and the comments entered by user. It holds the logic to display the right status in the dropdown list based on the current status.

2. __VerifyItems folder__
The components inside this folder handles the tables where the user can verify the products that were returned. It is available to be visited when request status is `pendingVerification`. When user hits saves, it sends a mutation passed down by the `ReturnDetailsProvider` to save the new status / comments in MD. 

3. __ReturnDetailsProvider.tsx__
- This provider fetches the return request details and pass it down via context API. It also aggregates the mutation to update the request, updating the cache right after the call.

### How to test it:

In [this workspace](https://rmav3updatestausui--powerplanet.myvtex.com/), create a new return request.
After doing so, go to the admin and find the request newly created. Visiting the page for the request, one can test the whole flow to update a request.

### CODE REVIEW

File: react/admin/ReturnDetails/ReturnDetailsContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          >
+            <FormattedMessage id="admin/return-app.return-request-details.error.description" />
+          </EmptyState>
+        ) : loading ? (
+          <Spinner />
+        ) : !data?.returnRequestDetails ? null : (
+          <>
+            {detailsPage !== 'return-details' ? null : (
+              <>
+                <div>Status {data.returnRequestDetails.status}</div>
```
Comment (Git-the-Sanz) [inline react/admin/ReturnDetails/ReturnDetailsContainer.tsx:62]:
```suggestion
                <div><FormattedMessage id="returns.status" /> {data.returnRequestDetails.status}</div>
```

Comment (Git-the-Sanz) [reply react/admin/ReturnDetails/ReturnDetailsContainer.tsx:62]: `"returns.status": "Status"`
Taken from messages

File: react/admin/ReturnDetails/ReturnDetailsContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          <EmptyState
+            title={
+              <FormattedMessage id="admin/return-app.return-request-details.error.title" />
+            }
+          >
+            <FormattedMessage id="admin/return-app.return-request-details.error.description" />
+          </EmptyState>
+        ) : loading ? (
+          <Spinner />
+        ) : !data?.returnRequestDetails ? null : (
```
Comment (Git-the-Sanz) [inline react/admin/ReturnDetails/ReturnDetailsContainer.tsx:58]: What do you think of making the returned components, at least from `data`, more readable? 
Maybe having an additional `wrapper` component for the Layout, or a functional component that accepts the props required for either page.

File: react/admin/ReturnDetails/components/VerifyItems/ProductActionStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  selectedQuantity,
+}: {
+  quantity: number
+  selectedQuantity: number
+}) => {
+  const actionStatus = getActionStatus({ quantity, selectedQuantity })
+
+  return (
+    <>
+      {actionStatus !== 'approve' ? null : (
```
Comment (Git-the-Sanz) [inline react/admin/ReturnDetails/components/VerifyItems/ProductActionStatus.tsx:31]: A nitpick, but same thing can be achieved with just &&
`actionStatus !== 'approve' && (<Component>)`

**Review by Git-the-Sanz — APPROVED** [review]
Good to go! 🚀

File: react/admin/ReturnDetails/ReturnDetailsContainer.tsx (hunk 2 above)
Comment (filafb) [reply react/admin/ReturnDetails/ReturnDetailsContainer.tsx]: Yeah. Logic was a little bit hard to follow. Sorry. I added a Loader component. Let me know what you think now.

File: react/admin/ReturnDetails/components/VerifyItems/ProductActionStatus.tsx (hunk 3 above)
Comment (filafb) [reply react/admin/ReturnDetails/components/VerifyItems/ProductActionStatus.tsx:31]: I stopped using `&&` because of this: https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx
